zenoh-kuksa-provider/src/main.rs

```diff
@@ -39,69 +42,59 @@ struct Args {
-    let config_str = fs::read_to_string(path)?;
-    let config = json5::from_str(&config_str)?;
+impl Args {
+    fn read_config(&self) -> Result<ProviderConfig, Box<dyn std::error::Error>> {
+        let config_str = fs::read_to_string(&self.config)?;
+        let config = json5::from_str(&config_str)?;
 
-    Ok(config)
+        Ok(config)
+    }
 }
 
-async fn handling_zenoh_subscribtion(
-    provider_config: Arc<ProviderConfig>,
+async fn handling_zenoh_subscription(
     session: Arc<Session>,
     metadata_store: MetadataStore,
     mut kuksa_client: kuksa::Client,
 ) {
-    info!("Listening on selector: {:?}", provider_config.zenoh.key_exp);
+    info!("Listening on selector: {:?}", VEHCILE_KEY_EXPR);
 
-    let provider_config_clone = Arc::clone(&provider_config);
-    let subscriber = session
-        .declare_subscriber(provider_config_clone.zenoh.key_exp.clone())
-        .res()
-        .await
-        .unwrap();
+    let subscriber = session.declare_subscriber(VEHCILE_KEY_EXPR).await.unwrap();
 
     let store = metadata_store.lock().await;
 
     while let Ok(sample) = subscriber.recv_async().await {
-        let vss_path = sample.key_expr.replace("/", ".").to_string();
+        let vss_path = sample.key_expr().replace("/", ".").to_string();
 
         debug!(
-            "Received ('{}': '{}' with attachment: {:?})",
-            &vss_path, sample.value, sample.attachment
+            "Received ('{}': '{:?}' with attachment: {:?})",
+            &vss_path,
+            sample.payload(),
+            sample.attachment()
         );
 
-        let field_type = extract_attachment_as_string(&sample);
-
-        if field_type == "currentValue" {
-            let datapoint_update = new_datapoint_for_update(&vss_path, &sample, &store);
-
-            debug!("Forwarding: {:#?}", datapoint_update);
-            kuksa_client
-                .set_current_values(datapoint_update)
-                .await
-                .unwrap();
+        if let Some(field_type) = extract_attachment_as_string(&sample) {
+            if field_type == "currentValue" {
+                let datapoint_update = new_datapoint_for_update(&vss_path, &sample, &store);
+                debug!("Forwarding: {:#?}", datapoint_update);
+                if let Err(e) = kuksa_client.set_current_values(datapoint_update).await {
+                    error!("failed to publish current value to Kuksa Databroker: {e}");
+                }
+            }
         }
     }
 }
 
 async fn publish_to_zenoh(
-    provider_config: Arc<ProviderConfig>,
+    provider_config: ProviderConfig,
     session: Arc<Session>,
     mut kuksa_client: kuksa::Client,
 ) {
-    let attachment = Some(String::from("type=targetValue"));
-
     let vss_paths = Vec::from_iter(provider_config.signals.iter().map(String::as_str));
 
-    let mut publishers: HashMap<String, zenoh::publication::Publisher> = HashMap::new();
+    let mut publishers: HashMap<String, Publisher<'_>> = HashMap::new();
```

Comment:
why change from explicitly telling what type you expect to `'_` ?

Reply:
Publisher expects a life time param, but does not require a type ...